src/transformers/models/mask2former/configuration_mask2former.py

```diff
@@ -0,0 +1,334 @@
+        self.encoder_feedforward_dim = encoder_feedforward_dim
+        self.norm = norm
+        self.encoder_layers = encoder_layers
+        self.decoder_layers = decoder_layers
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.dim_feedforward = dim_feedforward
+        self.pre_norm = pre_norm
+        self.enforce_input_proj = enforce_input_proj
+        self.common_stride = common_stride
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike], **kwargs) -> "PretrainedConfig":
+
+        config_dict, kwargs = cls.get_config_dict(pretrained_model_name_or_path, **kwargs)
+
+        # get the decoder config dict if we are loading from Mask2FormerConfig
+        if config_dict.get("model_type") == "mask2former":
+            config_dict = config_dict["decoder_config"]
+
+        if "model_type" in config_dict and hasattr(cls, "model_type") and config_dict["model_type"] != cls.model_type:
+            logger.warning(
+                f"You are using a model of type {config_dict['model_type']} to instantiate a model of type "
+                f"{cls.model_type}. This is not supported for all configurations of models and can yield errors."
+            )
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class Mask2FormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`Mask2FormerModel`]. It is used to instantiate a
+    Mask2Former model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the Mask2Former
+    [facebook/mask2former-instance-swin-small-coco](facebook/mask2former-instance-swin-small-coco) architecture trained
+    on [ADE20k-150](https://huggingface.co/datasets/scene_parse_150).
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Currently, Mask2Former only supports the [Swin Transformer](swin) as backbone.
+
+    Args:
+        backbone_config (`PretrainedConfig` or `dict`, *optional*, defaults to `SwinConfig()`):
+            The configuration of the backbone model. if unset, the configuration corresponding to
+            `swin-base-patch4-window12-384` will be used.
+        decoder_config (`dict`, *optional*):
+            The configuration passed to the pixel decoder and transformer decoder models. Includes the number of
+            layers, hidden state dimensions, normalization settings, etc.
+        ignore_value (`int`, *optional*, defaults to 255):
+            Category id to be ignored during training.
+        num_queries (`int`, *optional*, defaults to 150):
+            Number of queries
+        no_object_weight (`int`, *optional*, defaults to 0.1):
+            Weight to apply to the null (no object) class.
```

Comment:
```suggestion
            The weight to apply to the null (no object) class.
```
for consistency with the ones below